examples/jest_integration/worker.fuzz.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint no-undef: 0 */
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const target = require("./target");
+
+const startupTeardownCalls = [];
+
+const addCallLog = (uniqueId) => {
+	startupTeardownCalls.push(uniqueId);
+};
+
+beforeAll(() => {
+	return new Promise((resolve, reject) => {
+		setTimeout(() => {
+			addCallLog("Top-level beforeAll");
+			resolve(undefined);
+		}, 100);
+	});
+}, 1000);
+
+describe("My describe", () => {
+	beforeAll((done) => {
+		const callLog = "My describe: beforeAll";
+		addCallLog(callLog);
+		done();
+	});
+
+	afterAll((done) => {
+		addCallLog("My describe: afterAll");
+		done();
+	});
+
+	beforeEach(() => {
+		return new Promise((resolve, reject) => {
+			setTimeout(() => {
+				addCallLog("My describe: beforeEach");
+				resolve(undefined);
+			}, 100);
+		});
+	}, 200);
+
+	afterEach(() => {
+		let test = 0;
+		for (let i = 0; i < 1000; i++) {
+			test++; // Do nothing
+		}
+		addCallLog("My describe: afterEach");
+	});
+
+	it.fuzz("My regression test", async (data) => {
+		addCallLog("My fuzz test 1");
+		target.fuzzMe(data);
+	});
+
+	it.fuzz("My other regression test", (data) => {
+		addCallLog("My fuzz test 2");
+		target.fuzzMe(data);
+	});
+
+	describe("My nested describe", () => {
+		afterAll(() => {
```

Comment:
Order the `beforeXX` and `afterXX` methods like further up?

Reply:
That order was chosen on purpose. My reasoning was to test that even if beforeXX can be found after afterXX in code, the hook will run before the afterXX.

Reply:
Ah, makes sense :+1: